classes/Cart.php

```diff
@@ -202,6 +208,7 @@ public function __construct($id = null, $idLang = null)
         if (null !== $idLang) {
             $this->id_lang = (int) (Language::getLanguage($idLang) !== false) ? $idLang : Configuration::get('PS_LANG_DEFAULT');
         }
+        $this->id_lang_associated = $this->getAssociatedLanguage()->getId();
```

Comment:
This shouldn't be done in the constructor, but instead in the `getAssociatedLanguage` function itself, this way you only request the Language when it's needed. So instead of storing the `id_lang_associated` you should have an `$associated_language` field to store the object itself

Also as most cache strategies this could result in unexpected bugs, so we need a way to clear this cache or at least a parameter to force refreshing it So I suggest adding a `$forcerefresh` optional boolean parameter in the `getAssociatedLanguage(bool $forceRefresh = false)` function

Reply:
@jolelievre 
> Also as most cache strategies this could result in unexpected bugs

Can you really imagine when this could give a bug ? There is no way to change the language of a cart anyway (or where and when would this be done?

Reply:
@lmeyer1 but `getAssociatedLanguage` was introduced exactly to prevent errors after deleting a language, it's not a hypothetical use case, it was reported before :)

Reply:
Hi @lmeyer1 this is the original PR that introduced this method https://github.com/PrestaShop/PrestaShop/pull/22677

As @kpodemski this was not made just for fun but to fix an actual bug, you can get frustrated I understand but it doesn't mean we'll approve this faster ^^

Also when I meant it can be done more generic I meant that instead of modifying the Cart class you can handle your optimization in ObjectModel where the function is implemented instead (you'll see more details in my previous comments), so it's just about this language method but in the parent class.

Also, we try to anticipate the unexpected side effects, but as you said it should be fine as long as we are in the same process where there are few chances that the language is deleted in the meantime. I also thought this could impact the behat test that ensures this use case because they are all executed in the same process, but it seems to be fine, probably because each action (CQRS commands/queries) actually instantiates another Cart instance. As long as the tests are green it seems to be fine.

Reply:
@jolelievre Thanks for your explanations.

> you can get frustrated I understand but it doesn't mean we'll approve this faster ^^

At least I understand now what that pickyness is all about...